ingest/anndata_.py

```diff
@@ -35,3 +44,66 @@ def validate(self):
         except ValueError:
             return False
 
+    @staticmethod
+    def generate_cluster_header(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering NAME line to file
+        """
+        dim2 = ['NAME', 'X', 'Y']
+        dim3 = dim2 + ['Z']
+        clustering_dimension = adata.obsm[clustering_name].shape[1]
+        if clustering_dimension == 2:
+            headers = dim2
+        elif clustering_dimension == 3:
+            headers = dim3
+        elif clustering_dimension > 3:
+            msg = f"Too many dimensions for visualization in obsm \"{clustering_name}\", found {clustering_dimension}, expected 2 or 3."
+            raise ValueError(msg)
+        else:
+            msg = f"Too few dimensions for visualization in obsm \"{clustering_name}\", found {clustering_dimension}, expected 2 or 3."
+            raise ValueError(msg)
+        with open(f"{clustering_name}.tsv", "w") as f:
+            f.write('\t'.join(headers) + '\n')
+
+    @staticmethod
+    def generate_cluster_type_declaration(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering TYPE line to file
+        """
+        clustering_dimension = adata.obsm[clustering_name].shape[1]
+        types = ["TYPE", *["numeric"] * clustering_dimension]
+        with open(f"{clustering_name}.tsv", "a") as f:
```

Comment:
Non-blocking: I think it could be useful to have another segment to the filename to denote that these are interstitial files generated by this process that will make gathering them later in `files_to_delocalize` simpler.  Like `{clustering_name}.anndata_segment.tsv` or something.

Reply:
Great idea. For now, I've harded-coded two segments as `cluster.anndata_segment` to the file name (so we can grab all `anndata_segment` files, or just files of a particular type, by globbing)